src/modules/kvs/kvs.c

```diff
@@ -903,6 +911,28 @@ static void kvstxn_apply_cb (flux_future_t *f, void *arg)
     kvstxn_apply (kt);
 }
 
+static void start_sync_timer (struct kvs_ctx *ctx, struct kvsroot *root)
+{
+    if (ctx->rank == 0
+        && ctx->sync > 0.0
+        && root->is_primary
+        && !ctx->sync_started) {
+        /* monotime_since() returns in milliseconds, have to adjust to
+         * seconds */
+        double t = monotime_since (ctx->last_sync) / 1000.0;
+        double after = 0.0;
+        if (t < ctx->sync)
+            after = ctx->sync - t;
+        /* else we haven't synced in a long time, so set timer
+         * to set off immediately (i.e. after = 0.0)
+         */
```

Comment:
This seems a bit weird to force a sync on the first commit after the system has been idle, when this commit is likely to be immediately followed by others if it 's a job (the common case).  Maybe we could describe the sync parameter as "the maximum  length of time dirty KVS data will remain in memory before being flushed to the backing store" and then just set the timer to the fixed configured value?

That seems more intuitive to me.

Reply:
Is the comment perhaps misleading?  I initialize `last_sync` to when the module was loaded, so the first sync should occur "sync period" time after the first commit.

Reply:
Maybe I'm confused, or maybe I wasn't clear - I thought `t` was tracking the time since the last sync and would shorten the timer according to how long it's been since the last sync.   So for example the sync period is 30m and the system is idle for a day and then someone runs a job, the first commit would trigger a sync.  But since nothing else has been committed, there really isn't any increased urgency to do the sync right then.

If instead we think about the sync time period as placing an upper bound on how long after a change we wait to checkpoint it, then we would just start the timer with a fixed duration when ever something changes.

Reply:
ok, i see your point now.  So instead of "checkpoint every X time", you're thinking it should be "checkpoint if anything older than X time"